server/src/test/scala/org/http4s/server/middleware/AutoSlashSpec.scala

```diff
@@ -53,5 +55,26 @@ class AutoSlashSpec extends Http4sSpec with Http4sLegacyMatchersIO {
       router.orNotFound(Request[IO](uri = uri("/public/ping"))) must returnStatus(Status.Ok)
       router.orNotFound(Request[IO](uri = uri("/public/ping/"))) must returnStatus(Status.Ok)
     }
+
+    "Be created via httpRoutes constructor" in {
+      val req = Request[IO](uri = uri("/ping/"))
+      AutoSlash.httpRoutes(route).orNotFound(req) must returnStatus(Status.Ok)
+    }
+
+    "Be created via httpApp constructor" in {
+      val req = Request[IO](uri = uri("/ping/"))
+      val httpApp: HttpApp[IO] = HttpApp { r =>
+        import org.http4s.dsl.io._
+        if (r.pathInfo == "/ping") Ok()
+        else throw new Exception("Cannot handle request!")
+      }
+
+      implicit val invalidMonoidKIO: MonoidK[IO] = new MonoidK[IO] {
+        def combineK[A](x: IO[A], y: IO[A]): IO[A] = IO.ioSemigroupK.combineK[A](x, y)
+        override def empty[A]: IO[A] = IO.raiseError(new Exception("error"))
```

Comment:
This `MonoidK[IO]` instance doesn't feel to be right, it serves the purpose of the test, but I don't think it's correct.

Reply:
`Future.filter` implies something like this (it throws a `NoSuchElementException`), but I think is a wart in the standard library.  Requiring something like this is a great indication that what we're testing is not right.